src/DeviceInput/deviceInputSystem.ts

```diff
@@ -0,0 +1,223 @@
+    public onDeviceDisconnected(callback: (deviceName: string) => void): void {
+        this._onDeviceDisconnected = callback;
+    }
+
+    // Private functions
+    /**
+     * Add device and inputs to device map
+     * @param deviceName Assigned name of device (may be SN)
+     * @param numberOfInputs Number of input entries to create for given device
+     */
+    private registerDevice(deviceName: string, numberOfInputs: number) {
+        if (!this._inputs.has(deviceName)) {
+            var device: Array<number> = [];
+            for (var i = 0; i < numberOfInputs; i++) {
+                device.push(0);
+            }
+
+            this._inputs.set(deviceName, device);
+        }
+    }
+
+    /**
+     * Given a specific device name, remove that device from the device map
+     * @param deviceName Name of device to be removed
+     */
+    private deregisterDevice(deviceName: string) {
+        this._inputs.delete(deviceName);
+    }
+
+    /**
+     * Handle all actions that come from keyboard interaction
+     */
+    private handleKeyActions() {
+        window.addEventListener("keydown", (evt) => {
+            if (!this._keyboardActive) {
+                this._keyboardActive = true;
+                this.registerDevice(DeviceInputSystem.KEYBOARD_DEVICE, 222);
+                this._onDeviceConnected(DeviceInputSystem.KEYBOARD_DEVICE);
+            }
+
+            let kbKey = this._inputs.get(DeviceInputSystem.KEYBOARD_DEVICE);
+            if (kbKey) {
+                kbKey[evt.keyCode] = 1;
+            }
+        });
+
+        window.addEventListener("keyup", (evt) => {
+            let kbKey = this._inputs.get(DeviceInputSystem.KEYBOARD_DEVICE);
+            if (kbKey) {
+                kbKey[evt.keyCode] = 0;
+            }
+        });
+    }
+
+    /**
+     * Handle all actions that come from mouse interaction
+     */
+    private handleMouseActions() {
+        this._elementToAttachTo?.addEventListener("pointermove", (evt) => {
+            if (!this._mouseActive) {
+                this._mouseActive = true;
+                this.registerDevice(DeviceInputSystem.POINTER_DEVICE, 5);
+                this._onDeviceConnected(DeviceInputSystem.POINTER_DEVICE);
+            }
+
+            let mouseX = this._inputs.get(DeviceInputSystem.POINTER_DEVICE);
+            if (mouseX) {
+                mouseX[0] = evt.clientX;
+            }
+
+            let mouseY = this._inputs.get(DeviceInputSystem.POINTER_DEVICE);
+            if (mouseY) {
+                mouseY[1] = evt.clientY;
+            }
+        });
+
+        this._elementToAttachTo?.addEventListener("pointerdown", (evt) => {
+            if (!this._mouseActive) {
+                this._mouseActive = true;
+                this.registerDevice(DeviceInputSystem.POINTER_DEVICE, 5);
+                this._onDeviceConnected(DeviceInputSystem.POINTER_DEVICE);
+            }
+
+            let mouseButton = this._inputs.get(DeviceInputSystem.POINTER_DEVICE);

```

Comment:
I would call it pointerBUtton to be aligned with the event name

Reply:
Changed all mouse refs to pointer